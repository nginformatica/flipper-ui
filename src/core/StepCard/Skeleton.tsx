import { Box } from '@material-ui/core'
import MuiExpansionPanel from '@material-ui/core/ExpansionPanel'
import { Skeleton } from '@material-ui/lab'
import React from 'react'
import './colors.css'
import {
    Container,
    NormalProgressContainer,
    StepCardColumn,
    StepCardRow,
    StepContainer,
    TitleContainer
} from './styles'

interface IStepCardPanelProps {
    expandable: boolean
    showBottomPercentage: boolean
    showIcon: boolean
    subTitle?: boolean
}

const StepCardSkeleton = (props: IStepCardPanelProps) => {
    const { expandable, showBottomPercentage, showIcon, subTitle } = props

    const TextSkeleton = () => (
        <Skeleton variant='text' style={{ minWidth: '400px' }} height={40} />
    )

    const NormalProgress = () => (
        <NormalProgressContainer>
            <Skeleton
                variant='text'
                height={40}
                style={{ minWidth: '100px', margin: '4px 0px' }}
            />
            <Box style={{ width: '100%' }}>
                <Skeleton variant='text' />
            </Box>
        </NormalProgressContainer>
    )

    return (
        <Container>
            <MuiExpansionPanel>
                <StepContainer withPadding={!expandable}>
                    <StepCardRow height={expandable ? '100px' : '100%'}>
                        <StepCardColumn justifyContent='start'>
                            {showIcon && (
                                <Skeleton
                                    variant='circle'
                                    width={40}
                                    height={40}
                                />
                            )}
                            <TitleContainer>
                                <TextSkeleton />
                                {subTitle && <TextSkeleton />}
                            </TitleContainer>
                        </StepCardColumn>
                        <StepCardColumn justifyContent='end'>
                            {expandable ? <TextSkeleton /> : <NormalProgress />}
                        </StepCardColumn>
                    </StepCardRow>
                    {showBottomPercentage && (
                        <StepCardRow height='15px'>
                            <Box style={{ width: '100%' }}>
                                <Skeleton variant='text' />
                            </Box>
                        </StepCardRow>
                    )}
                </StepContainer>
            </MuiExpansionPanel>
        </Container>
    )
}

export default StepCardSkeleton
