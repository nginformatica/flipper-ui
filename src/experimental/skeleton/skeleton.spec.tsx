import * as React from 'react'
import { render, screen } from '@testing-library/react'
import Skeleton from '.'

const Simple = () => <Skeleton />
const WithCustomHeight = () => <Skeleton width={64} height={64} />
const Circle = () => <Skeleton circle width={64} height={64} />

describe('Skeleton', () => {
    it('should render component', () => {
        render(<Simple />)

        const loading = screen.getByTestId('skeleton-container')

        expect(loading).toBeDefined()
    })

    it('should match snapshot - Simple', () => {
        const container = render(<Simple />)

        expect(container).toMatchSnapshot()
    })

    it('should match snapshot - WithCustomHeight', () => {
        const container = render(<WithCustomHeight />)

        expect(container).toMatchSnapshot()
    })

    it('should match snapshot - Circle', () => {
        const container = render(<Circle />)

        expect(container).toMatchSnapshot()
    })
})
