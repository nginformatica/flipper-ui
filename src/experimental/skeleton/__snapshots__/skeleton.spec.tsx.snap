// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton should match snapshot - Circle 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        aria-busy="true"
        aria-live="polite"
        data-testid="skeleton-container"
      >
        <span
          class="react-loading-skeleton"
          style="width: 64px; height: 64px; border-radius: 50%; --base-color: #eeeeee; --highlight-color: #e0e0e0;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      aria-busy="true"
      aria-live="polite"
      data-testid="skeleton-container"
    >
      <span
        class="react-loading-skeleton"
        style="width: 64px; height: 64px; border-radius: 50%; --base-color: #eeeeee; --highlight-color: #e0e0e0;"
      >
        ‌
      </span>
      <br />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Skeleton should match snapshot - Simple 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        aria-busy="true"
        aria-live="polite"
        data-testid="skeleton-container"
      >
        <span
          class="react-loading-skeleton"
          style="--base-color: #eeeeee; --highlight-color: #e0e0e0;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      aria-busy="true"
      aria-live="polite"
      data-testid="skeleton-container"
    >
      <span
        class="react-loading-skeleton"
        style="--base-color: #eeeeee; --highlight-color: #e0e0e0;"
      >
        ‌
      </span>
      <br />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Skeleton should match snapshot - WithCustomHeight 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        aria-busy="true"
        aria-live="polite"
        data-testid="skeleton-container"
      >
        <span
          class="react-loading-skeleton"
          style="width: 64px; height: 64px; --base-color: #eeeeee; --highlight-color: #e0e0e0;"
        >
          ‌
        </span>
        <br />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      aria-busy="true"
      aria-live="polite"
      data-testid="skeleton-container"
    >
      <span
        class="react-loading-skeleton"
        style="width: 64px; height: 64px; --base-color: #eeeeee; --highlight-color: #e0e0e0;"
      >
        ‌
      </span>
      <br />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
